src/ln/functional_tests.rs

```diff
@@ -1449,6 +1449,153 @@ fn channel_reserve_test() {
 	do_channel_reserve_test(true);
 }
 
+#[test]
+fn channel_reserve_in_flight_removes() {
+	// In cases where one side claims an HTLC, it thinks it has additional available funds that it
+	// can send to its counterparty, but due to update ordering, the other side may not yet have
+	// considered those HTLCs fully removed.
+	// This tests that we don't count HTLCs which will not be included in the next remote
+	// commitment transaction towards the reserve value (as it implies no commitment transaction
+	// will be generated which violates the remote reserve value).
+	// This was broken previously, and discovered by the chanmon_fail_consistency fuzz test.
+	// To test this we: (XXX: Simplify me?)
+	//  * route two HTLCs from A to B,
+	//  * start routing a third HTLC from A to B,
+	//  * claim the first two HTLCs (though B will generate an update_fulfill for one, and put
+	//    the other claim in its holding cell, as it immediately goes into AwaitingRAA),
+	//  * deliver the first fulfill from B
+	//  * deliver the update_add and an RAA from A, resulting in B freeing the second holding cell
+	//    claim,
+	//  * deliver A's response CS and RAA.
+	//    This results in A having the second HTLC in AwaitingRemovedRemoteRevoke, but B having
+	//    removed it fully. B now has the push_msat plus the first two HTLCs in value.
+	//  * Now B happily sends another HTLC, potentially violating its reserve value from A's point
+	//    of view (if A counts the AwaitingRemovedRemoteRevoke HTLC).
+	let mut nodes = create_network(2);
+	let chan_1 = create_announced_chan_between_nodes(&nodes, 0, 1);
+
+	let b_chan_values = get_channel_value_stat!(nodes[1], chan_1.2);
+	// Route the first two HTLCs. Note that the test is to check that, when you consider the values
```

Comment:
Well IMO this note is simply explaining the test better than the long description above, maybe it should be in top-level comment

Reply:
Done.